lib/diagnostic.js

```diff
@@ -6,29 +6,55 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  let words = paragraph.split(/\s+/);
+  for (let i = 0; i < words.length; i++) {
+  let word = words[i].replace(/\W+/, '');
+  normalizeWords[i] = word.toUpperCase();
+}
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (normalizedWords) {
+  let uniqueWordsAsKeys = {};
+for (let i = 0; i < normalizedWords.length; i++) {
+  let word = normalizedWords[i];
+  uniqueWordsAsKeys[word] = true;
+}
 };
 
 // 3. Write a function that takes a string argument and returns the count of
 // words in a string.
 // Provide the *option* to count unique words instead of total words, by passing
 // a second argument.
 
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function (str) {
+  return str.split(" ").length;
+};
```

Comment:
Close, but not quite. This actually much easier than maybe what you thought. The `normalizeWords` function already gives you an array of words, so all you need to do is get its `length`.